lib/bolt/inventory/inventory2.rb

```diff
@@ -56,13 +54,24 @@ def target_names
       alias node_names target_names
 
       def get_targets(targets)
-        targets = expand_targets(targets)
-        targets = if targets.is_a? Array
-                    targets.flatten.uniq(&:name)
-                  else
-                    [targets]
-                  end
-        targets.map { |t| update_target(t) }
+        flat_target_list(targets).map { |t| update_target(t) }
+      end
+
+      def get_target(target)
+        # Used to track if new target was created
+        existing_targets = @targets.keys
+        target_array = flat_target_list(target)
+        if target_array.count > 1
+          matches = target_array.map(&:name)
+          raise ValidationError.new("'#{target}' refers to #{matches.count} targets", nil)
+        end
+        single_target = target_array.first
+        # add_to_group expect array
```

Comment:
You could theoretically just use `target_array` instead then, since it is already known to be exactly `[single_target]`

Reply:
I was thinking `single_target` was more readable than `target_array.first` everywhere.